pkg/apis/storage/validation/validation.go

```diff
@@ -430,7 +430,6 @@ func ValidateCSIDriverUpdate(new, old *storage.CSIDriver) field.ErrorList {
 	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(new.Spec.FSGroupPolicy, old.Spec.FSGroupPolicy, field.NewPath("spec", "fsGroupPolicy"))...)
 	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(new.Spec.PodInfoOnMount, old.Spec.PodInfoOnMount, field.NewPath("spec", "podInfoOnMount"))...)
 	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(new.Spec.VolumeLifecycleModes, old.Spec.VolumeLifecycleModes, field.NewPath("spec", "volumeLifecycleModes"))...)
-	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(new.Spec.StorageCapacity, old.Spec.StorageCapacity, field.NewPath("spec", "storageCapacity"))...)
```

Comment:
The comment in types.go should also be updated: https://github.com/kubernetes/kubernetes/blob/1f8b098e5af29392f1cb9af808a8bf915877b67f/pkg/apis/storage/types.go#L360

Reply:
This comment is still outstanding

Reply:
Oops, sorry for that. Instead of simply dropping the line, I decided to include historic information to avoid the mistake mentioned by Jordan (developer develops and tests with 1.23, then updating the field fails with older Kubernetes):

```
This field was immutable in Kubernetes <= 1.22 and now is mutable.
```